components/automate-cli/cmd/chef-automate/certRotate.go

```diff
@@ -574,6 +647,90 @@ func (c *certRotateFlow) copyAndExecute(ips []string, sshUtil SSHUtil, timestamp
 	return nil
 }
 
+// copyAndExecute will copy the toml file to each required node and then execute the set of commands.
+func (c *certRotateFlow) copyAndExecuteConcurrently(nu NodeOpUtils, ips []string, sshUtil SSHUtil, timestamp string, remoteService string, fileName string, scriptCommands string, flagsObj *certRotateFlags, printCertRotateOutput func(CertRotateCmdResult, string, *cli.Writer)) error {
+	//var err error
+	var tomlFilePath string
+	responseChan := make(chan CertRotateCmdResult, len(ips))
+	defer close(responseChan)
+	for i := 0; i < len(ips); i++ {
+		infra, _, err := nu.getHaInfraDetails()
+		if err != nil {
+			return err
+		}
+		SSHConfig := c.getSshDetails(infra)
+		SSHUtils := NewSSHUtil(SSHConfig)
+		SSHConfig.timeout = flagsObj.timeout
+		SSHUtils.setSSHConfig(SSHConfig)
+		go func(hostIP string, responseChan chan CertRotateCmdResult) {
+			rc := CertRotateCmdResult{hostIP, "", nil, writer, remoteService}
+			SSHUtils.getSSHConfig().hostIP = hostIP
+			if (flagsObj.postgres || flagsObj.opensearch) && remoteService != "frontend" {
+				tomlFilePath, err = c.getMerger(fileName, timestamp, remoteService, GET_USER_CONFIG, SSHUtils)
+				if err != nil {
+					rc.err = err
+					responseChan <- rc
+					return
+				}
+				// Copying the new toml file which includes both old and new configurations (for backend nodes).
+				err = SSHUtils.copyFileToRemote(tomlFilePath, remoteService+timestamp, false)
```

Comment:
Maybe we should name the function as `copyAndExecuteConcurrentlyToFrontEndNodes` and remove the checks for backend nodes in this function. Let discuss this and take a call

Reply:
changed the func name and removed the checks for backend nodes.